### PR TITLE
Refactor blocks

### PR DESCRIPTION
**Pull request by debauchery1st**
# Description

Fixes # code climate - Function `AddTransaction` has 305 lines of code

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Change Status

- [x] Complete, tested, ready to review and merge

# How Has This Been Tested?

- [x] components function as expected.

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

**Comment by debauchery1st** [comment]
[code climate - Function `AddTransaction` has 305 lines of code](https://trello.com/c/9wweniss/66-code-climate-function-addtransaction-has-305-lines-of-code)

File: src/components/Modal_Components/AddTransaction.jsx [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import React, { useState, useEffect } from "react";
-import { Back_Continue } from "./Back_Continue";
-import { Modal_Title } from "./Modal_Title";
-import MuiDialogActions from "@material-ui/core/DialogActions";
```
Comment (israelgonzalezb) [inline src/components/Modal_Components/AddTransaction.jsx]: Good job at catching the weird casing of these imports.

File: src/components/Modal_Components/AddTransaction.jsx [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
+import BackContinue from "./BackContinue";
+import ModalTitle from "./ModalTitle";
+// import MuiDialogActions from "@material-ui/core/DialogActions";
```
Comment (israelgonzalezb) [inline src/components/Modal_Components/AddTransaction.jsx]: Let's delete these unused imports

File: src/components/Modal_Components/AddTransaction.jsx [hunk 3]
```diff
@@ -31,12 +31,12 @@ const DialogContent = withStyles(theme => ({
...
-    margin: 0,
-    padding: theme.spacing(1)
-  }
-}))(MuiDialogActions);
+// const DialogActions = withStyles(theme => ({
+//   root: {
+//     margin: 0,
+//     padding: theme.spacing(1)
+//   }
+// }))(MuiDialogActions);
```
Comment (israelgonzalezb) [inline src/components/Modal_Components/AddTransaction.jsx]: Can we delete this?

File: src/components/Modal_Components/BackContinue.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
```
Comment (israelgonzalezb) [inline src/components/Modal_Components/BackContinue.js]: This file is showing as if it's brand new but it's just a side effect of fixing the import casing.

File: src/components/Modal_Components/demomanualBlocks.js [hunk 5]
```diff
@@ -1,383 +1,156 @@
...
-const useStyles = makeStyles(theme => ({
-  root: {
-    width: "100%",
-    display: "flex",
-    justifyContent: "space-between"
-  },
-  formControl: {
-    width: "45%"
-  }
-}));
```
Comment (israelgonzalezb) [inline src/components/Modal_Components/demomanualBlocks.js]: Good idea of extracting this into a separate file.

File: src/components/Modal_Components/demomanualBlocks.js [hunk 6]
```diff
@@ -1,383 +1,156 @@
...
+    editBlock: {
+      name: "",
+      budget: null,
+      categoryid: null,
+      total: 0
+    },
+    edit: true,
+    failedEdit: false,
+    failedCustom: false
+  });
```
Comment (israelgonzalezb) [inline src/components/Modal_Components/demomanualBlocks.js]: Refactored all those state hooks into one, awesome!

File: src/components/Modal_Components/demomanualBlocks.js [hunk 7]
```diff
@@ -1,383 +1,156 @@
...
-const handleBudget = e => {
-    setEditBlock({...editBlock,budget:e.target.value})
-    setCustomBlock({...customBlock,budget:e.target.value})
-    console.log(editBlock)
-    console.log(customBlock)
-}
-
-const handleSubmit = e => {
-    e.preventDefault()
-    if (customBlock.name === null || isNaN(customBlock.budget) || !isNaN(customBlock.name)){
```
Comment (israelgonzalezb) [inline src/components/Modal_Components/demomanualBlocks.js]: Good job at pulling this out into a variable on 67-70. Made it much more readable than this giant conditional line.

File: src/components/Modal_Components/dialogViews/dialogComponents.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import ModalTitle from "../ModalTitle";
```
Comment (israelgonzalezb) [inline src/components/Modal_Components/dialogViews/dialogComponents.js]: Can we add a comment to explain why/how this is used?

File: src/components/Modal_Components/dialogViews/dialogComponents.js [hunk 9]
```diff
@@ -5,7 +5,16 @@ import Dialog from "@material-ui/core/Dialog";
...
+/**
+ * @description dialogComponents
+ *
+ * provides one local import containing components common to the dialogViews.
+ *
+ * helps ease the transition from one component framework to another.
+ *
+ * example:
+ *    import {TextField, Button } from "./dialogComponents";
+ */
```
Comment (israelgonzalezb) [inline src/components/Modal_Components/dialogViews/dialogComponents.js]: Perfect 👌

**Comment by israelgonzalezb** [comment]
Awesome job dude! High quality work

**Pull request merged**